core/src/main/java/io/cucumber/core/options/CucumberMessageStoreUrl.java

```diff
@@ -0,0 +1,28 @@
+package io.cucumber.core.options;
+
+import io.cucumber.core.exception.CucumberException;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+
+public class CucumberMessageStoreUrl {
+    public static String getPluginString() {
+        String url = getUrl();
+        try {
+            String apiUrl = new URL(new URL(url), "/api/reports").toExternalForm();
+            return String.format("message:%s", apiUrl);
+        } catch (MalformedURLException e) {
+            throw new CucumberException("Bad URL: " + url);
+        }
+    }
+
+    public static String getUrl() {
+        Map<String, String> props = CucumberProperties.create();
+        String url = props.get("CUCUMBER_MESSAGE_STORE_URL");
```

Comment:
Please also register this property in the `Constants` class.

Reply:
Comes to mind that this should probably also be written property style. The `CucumberProperties` class will translate this to upper case underscores.